bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,234 @@
+                    if (YIOremoteDockwebSocket_ClientSocket.get_boolean_authentication_required()) {
+                        logger.debug("send authentication to YIO dock");
+                        YIOremoteDockwebSocket_ClientSocket.sendMessage(YIOREMOTEMESSAGETYPE.AUTHENTICATE,
+                                config.yiodockaccesstoken);
+                        Thread.sleep(1000);
+
+                        if (YIOremoteDockwebSocket_ClientSocket.get_boolean_authentication_ok()) {
+                            YIOREMOTEHANDLESTATUS_actualstatus = YIOREMOTEHANDLESTATUS.AUTHENTICATED;
+
+                        } else {
+                            logger.debug("authentication to YIO dock not ok");
+                            YIOREMOTEHANDLESTATUS_actualstatus = YIOREMOTEHANDLESTATUS.AUTHENTICATED_FAILED;
+                        }
+                    } else {
+                        logger.debug("authentication error YIO dock");
+                    }
+
+                } catch (IllegalArgumentException e) {
+                    logger.warn("JSON convertion failure {}", e.toString());
+                }
+
+            } catch (
+
+            Exception e) {
+                logger.warn("Web socket connect failed {}", e.toString());
+                // throw new IOException("Web socket start failed");
+            }
+        } catch (URISyntaxException e) {
+            logger.debug("Initialize web socket failed {}", e.toString());
+        }
+
+        if (YIOREMOTEHANDLESTATUS_actualstatus.equals(YIOREMOTEHANDLESTATUS.AUTHENTICATED)) {
+            updateStatus(ThingStatus.ONLINE);
+            Runnable heartbeatpolling = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        if (YIOREMOTEHANDLESTATUS_actualstatus.equals(YIOREMOTEHANDLESTATUS.AUTHENTICATED)) {
+                            YIOremoteDockwebSocket_ClientSocket.sendMessage(YIOREMOTEMESSAGETYPE.HEARTBEAT, "");
+
+                            Thread.sleep(1000);
+
+                            if (YIOremoteDockwebSocket_ClientSocket.get_boolean_heartbeat()) {
+                                logger.debug("heartbeat ok");
+                                updateChannelString(GROUP_OUTPUT, YIODOCKSTATUS,
+                                        YIOremoteDockwebSocket_ClientSocket.get_string_receivedstatus());
+                            } else {
+                                logger.warn("Connection lost no ping from YIO DOCK");
+                                YIOREMOTEHANDLESTATUS_actualstatus = YIOREMOTEHANDLESTATUS.CONNECTION_FAILED;
+                                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                        "Connection lost no ping from YIO DOCK");
+                                updateChannelString(GROUP_OUTPUT, YIODOCKSTATUS, "Connection/Configuration Error");
+                                pollingJob.cancel(true);
```

Comment:
You could store `pollingJob` to a local variable an do a null check on that, to eliminate the compiler warning.